build.gradle

```diff
@@ -217,6 +218,18 @@ testClusters.integTest {
     systemProperty("java.library.path", "$rootDir/jni/release")
 }
 
+task integTestRemote(type: RestIntegTestTask) {
+    testClassesDirs = sourceSets.test.output.classesDirs
+    classpath = sourceSets.test.runtimeClasspath
+
+    // Run tests with remote cluster only if rest case is defined
+    if (System.getProperty("tests.rest.cluster") != null) {
+        filter {
+            includeTestsMatching "org.opensearch.knn.*IT"
```

Comment:
I think, you should add this in the filter to fix the BWC issue which Jack mentioned in review comments
excludeTestsMatching "org.opensearch.knn.bwc.*IT"

Reply:
Reference - https://github.com/opensearch-project/k-NN/blob/main/build.gradle#L176

Reply:
Thanks Naveen, it's a good point as we need to have multiple external clusters for BWC. For now I've added bwc tests to exclusion matching rule